clang/lib/Sema/AnalysisBasedWarnings.cpp

```diff
@@ -2987,6 +2987,7 @@ void clang::sema::AnalysisBasedWarnings::IssueWarnings(
   AC.getCFGBuildOptions().AddInitializers = true;
   AC.getCFGBuildOptions().AddImplicitDtors = true;
   AC.getCFGBuildOptions().AddTemporaryDtors = true;
+  AC.getCFGBuildOptions().AddLifetime = true;
```

Comment:
Since only one analysis needs this I wonder if we want to guard this by the condition that lifetime checks are enabled.

Reply:
\+1. You can use do this behind the boolean `EnableLifetimeSafetyAnalysis.`